src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1181,13 +1180,16 @@ CheckCitusVersion(int elevel)
  * this function logs an error with the specified elevel and returns false,
  * otherwise it returns true.
  */
-static bool
+bool
 CheckAvailableVersion(int elevel)
 {
 	char *availableVersion = NULL;
 
-	Assert(CitusHasBeenLoaded());
-	Assert(EnableVersionChecks);
+	if (citusVersionKnownCompatible ||
```

Comment:
Why are we checking the cache here?  It seems like it'd be better if we avoided that - in that case it seems like we could avoid resetting citusVersionKnownCompatible in multi_utility and make it private.  That seems like it'd be a significant improvement.